config/services.yaml

```diff
@@ -0,0 +1,201 @@
+        public: true
+        class: App\Util\UserManipulator
+        arguments:
+            - '@fos_user.user_manager'
+            - '@fos_user.util.token_generator'
+            - '@event_dispatcher'
+            - '@request_stack'
+
+    App\Form\Handler\OAuthRegistrationFormHandler:
+        public: true
+        class: App\Form\Handler\OAuthRegistrationFormHandler
+        arguments: ['@fos_user.user_manager', '@fos_user.util.token_generator']
+
+    App\Form\Type\OAuthRegistrationFormType:
+        tags:
+            - { name: form.type, alias: packagist_oauth_user_registration }
+
+    packagist.oauth.registration_form:
+        public: true
+        factory: ['@form.factory', create]
+        class: Symfony\Component\Form\Form
+        arguments:
+            - 'App\Form\Type\OAuthRegistrationFormType'
+
+    App\Model\PackageManager:
+        bind:
+            $options: { from: '%env(APP_MAILER_FROM_EMAIL)%', fromName: '%env(APP_MAILER_FROM_NAME)%' }
+
+    packagist.profile.form.type:
+        public: true
+        class: App\Form\Type\ProfileFormType
+        arguments: ['%fos_user.model.user.class%']
+        tags:
+            - { name: form.type, alias: packagist_user_profile }
+
+    App\Menu\MenuBuilder:
+        arguments: ['@knp_menu.factory', '@security.token_storage', '@translator']
+
+    packagist.menu.user:
+        class: Knp\Menu\MenuItem
+        factory: ['@App\Menu\MenuBuilder', createUserMenu]
+        tags:
+            - { name: knp_menu.menu, alias: user_menu }
+
+    packagist.menu.profile:
+        class: Knp\Menu\MenuItem
+        factory: ['@App\Menu\MenuBuilder', createProfileMenu]
+        tags:
+            - { name: knp_menu.menu, alias: profile_menu }
+
+    Algolia\AlgoliaSearch\SearchClient:
+        arguments: ['%env(ALGOLIA_APP_ID)%', '%env(ALGOLIA_ADMIN_KEY)%']
+        factory: ['Algolia\AlgoliaSearch\SearchClient', create]
+
+    App\Service\QueueWorker:
+        arguments:
+            - "@snc_redis.default_client"
+            - "@doctrine"
+            - "@logger"
+            - 'package:updates': '@App\Service\UpdaterWorker'
+              'githubuser:migrate': '@App\Service\GitHubUserMigrationWorker'
+              'security:advisory': '@App\Service\SecurityAdvisoryWorker'
+
+    App\Security\TwoFactorAuthManager:
+        public: true
+        class: App\Security\TwoFactorAuthManager
+        arguments:
+            - '@doctrine'
+            - '@mailer'
+            - '@twig'
+            - '@logger'
+            - '@session.flash_bag'
+            - { from: '%env(APP_MAILER_FROM_EMAIL)%', fromName: '%env(APP_MAILER_FROM_NAME)%' }
+
+    App\Security\TwoFactorAuthRateLimiter:
+        tags:
+            - { name: kernel.event_subscriber }
+
+    App\Security\BruteForceLoginFormAuthenticator:
+        bind:
+            $recaptchaVerifier: "@beelab_recaptcha2.verifier"
+
+    App\Security\RecaptchaHelper:
+        bind:
+            $recaptchaEnabled: "%env(bool:APP_RECAPTCHA_ENABLED)%"
```

Comment:
this could be added to the file-level default bindings, avoiding the need to define the services explicitly.